server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java

```diff
@@ -100,10 +108,81 @@ private Mail cloneMail(Mail mail) throws MessagingException {
         public MailQueueItem deQueue() throws MailQueueException, InterruptedException {
             while (true) {
                 MemoryMailQueueItem item = mailItems.takeLast();
+                inProcessingMailItems.add(item);
                 return mailQueueItemDecoratorFactory.decorate(item);
             }
         }
 
+        @Override
+        public long getSize() throws MailQueueException {
+            return mailItems.size() + inProcessingMailItems.size();
+        }
+
+        @Override
+        public long flush() throws MailQueueException {
+            return 0;
```

Comment:
why it is in this interface ?

Reply:
Good question. I guess it is a capability of Manageable mail queue, as Delay is **by default part of the MailQueue** interface. Seems logical.

The same way MemoryMailQueue do not implement delays, Memory mail queue do not implement flush.

IMO fair enough.